### DISCUSSION TITLE
Unmounting & remounting a script tag in <Head> causes the file to get re-downloaded & executed

### DISCUSSION THREAD

**Discussion by ericbiewener** [vercel/next.js#43514] [Ideas]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14.16.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

N/A, running locally

### Describe the Bug

Unmounting and then re-mounting a component that renders `<Head><script ...></Head>` will cause the `script` to re-download and `re-execute`.

### Expected Behavior

`<script>` tags should never get unmounted, and thus never re-downloaded nor re-executed.

At least, I think this is how it should work? Perhaps there are use cases I'm not aware of, but I can't think of a situation in which unmounting a script from the page really makes sense.

### To Reproduce

[Demo repo](https://github.com/ericbiewener/next-head-remount). Essential code:

```tsx
import Head from "next/head";
import { useState } from "react";

const HeadWithScript = () => (
  <Head>
    <script async src="/test-script.js" key="/test-script.js"></script>
  </Head>
);

export default function Home() {
  const [isMounted, setIsMounted] = useState(false);

  return (
    <>
      {isMounted && <HeadWithScript />}
      <div>Open your console and start toggling!</div>
      <br />
      <div>
        <button onClick={() => setIsMounted(!isMounted)}>
          Toggle script mounting
        </button>{" "}
        <span>Script is {isMounted ? "mounted" : "not mounted"}</span>
      </div>
    </>
  );
}
```

**test-script.js**
```js
console.log('Test script executed!')
```

Every time you re-mount the script, you'll see that log in the browser's console.

**Comment by ericbiewener** [comment]
Assuming there _is_ a valid use-case for redownloading/executing the script, then perhaps this should be a feature request instead, e.g.

```tsx
<Head permanent>
  <script ... />
</Head>
```

**Comment by jankaifer** [comment]
I'll convert this into the discussion since it's a better place for this question.
